### ISSUE TITLE
unable to update DOI due to creator field metadata

### ISSUE THREAD

**Issue by daslerr** [datacite/datacite#475]
Users report being unable to update DOIs via the Form in DOI Fabrica. They can make their changes, but clicking the Update DOI button does nothing. The creator metadata displays name tags (see screenshot). They did not enter the metadata this way, because the DOIs were originally created using the Form with manual entry. 
![doi issues](https://user-images.githubusercontent.com/3181623/44582948-94a2b880-a7a3-11e8-8b9c-2ed6e86b7698.PNG)

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_debn)

**Comment by richardhallett** [comment]
I can not reproduce this bug, it's obvious that the format for the creators field is incorrect and I expect there is a validation problem, but without being able to reproduce or additional examples hard to narrow down.
Closing for now, please re-open if you have further information or other examples.

**Comment by richardhallett** [comment]
Re-opening as there are still reports of this coming in.

**Comment by richardhallett** [comment]
Closing as still unable to re-reproduce, attempted various scenarios manually but to no avail.
Original example appears no longer to be a problem either.

**Comment by richardhallett** [comment]
This has been reported, so far I've not reproduced this exactly, however using a users metadata schema 3 on the form, this seems to clear the form entirely, I am wondering if it's a metadata schema version specific problem.

**Comment by richardhallett** [comment]
Looking into this further it appeared to be the stringify of JSON data and various serializing and deserializing, this would in some cases just end up causing a recursion of serializing/deserializing resulting in invalid JSON within the creators field.

I've solved this by not serialising to json and leaving that up to the ember data serializers, instead we just treat the Author an as an Array of authors, this is then parsed from new lines into appropriate array/object structure.

Side bonus, I also fixed a side issue with the fact the names would get re-arranged during this process.

This has been deployed to our test infrastructure.

**Comment by mfenner** [comment]
This looks good, but removes one important functionality: it strips out information that can not be edited in the form, e.g. ORCID ID. But this solution is cleaner, and we will have better form support for authors going forward.

**Comment by daslerr** [comment]
This has cropped up again, but this time the creator field is showing [object Object] when the user attempts to update the DOI record. It appears the creator field is not handling invalid multiline creators, such as FirstName LastName as shown below. 

![image](https://user-images.githubusercontent.com/3181623/47141896-f8d48b00-d2c1-11e8-9ca5-eb2c213820d5.png)

**Comment by richardhallett** [comment]
This has also cropped up with older schema versions, namely those that do not support familyName, givenName as attributes: 
See http://datacite.org/schema/kernel-2.2

**Comment by richardhallett** [comment]
So there are a couple things to this issue, sometimes the appearance of Object Object in the field is not actually due to the Creators field having a bug (i.e. in Bracco) but instead due to the API request to save which causes the form to bail out.
Ideally Fabrica should handle this better but I think that will be something we consider when we want to add more fields to Fabrica / extend the form.

To the above mentioned problems, I fixed the related "older schema" version, this was actually as mentioned a bug in the API causing it to break.

The multline does work, and I've tested this with various combinations.
